### PR TITLE
[Distributed] Another fix for getting required protocols for SR

### PR DESCRIPTION
**Pull request by ktoso**
A follow up to https://github.com/apple/swift/pull/69501 -- thanks @slavapestov for the review 🙏 

 rdar://116261701

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test linux

**Comment by ktoso** [comment]
@swift-ci Please clean test Linux platform

File: lib/AST/DistributedDecl.cpp [hunk 1]
```diff
@@ -119,15 +119,8 @@ Type swift::getSerializationRequirementTypesForMember(
...
-    for (auto requirement: signature.getRequirements()) {
-      if (requirement.getFirstType()->isEqual(SerReqAssocType) &&
-          requirement.getKind() == RequirementKind::Conformance) {
-        if (auto nominal = requirement.getSecondType()->getAnyNominal()) {
-          if (auto protocol = dyn_cast<ProtocolDecl>(nominal)) {
-            serializationRequirements.insert(protocol);
-          }
-        }
-      }
+    for (auto proto: signature->getRequiredProtocols(SerReqAssocType)) {
```
Comment (slavapestov) [inline lib/AST/DistributedDecl.cpp:122]: Thank you so much for doing this along with the other cleanups. The code in this file is much easier to follow now!

**Review by slavapestov — APPROVED** [review]

**Comment by ktoso** [comment]
@swift-ci please smoke test macos

**Comment by ktoso** [comment]
@swift-ci please smoke test linux

**Pull request merged**